src/csharp/Microsoft.Spark/Sql/CastUnpickledItems.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Spark.Sql
+{
+    /// <summary>
+    /// Cast UnpickledItems.
+    /// </summary>
+    internal static class CastUnpickledItems
+    {
+        /// <summary>
+        /// Cast unpickledItems from arraylist to the appropriate array.
+        /// </summary>
+        /// <param name="unpickledItems">Unpickled items that contains simple array</param>
+        /// <returns>Simple array after casting</returns>
+        public static object[] CastToSimpleArray(object unpickledItems)
```

Comment:
Looks like you have a case for 2d jagged arrays.  What about 3,4,..,n d arrays ?

Reply:
Will check and add the support for n d arrays

Reply:
Added support for n d arrays.